src/mapping_navigation/scripts/Models.py

```diff
@@ -163,6 +165,11 @@ def convert_path(self, index) -> List[Tuple[float, float]]:
 
         # Iterate through points in the path
         for con, p in sel_path.connections:
-            airsim_path.append(self.convert_point(p))
+            converted_point = self.convert_point(p)
+            if con.from_seg_id is not None:
+                segment_type = self.road_segments[con.from_seg_id].seg_type
+            else:
+                segment_type = RoadSegmentType.STRAIGHT
+            airsim_path.append((converted_point[0], converted_point[1], segment_type))

```

Comment:
is segment_type being used outside the scope it's declared in? If that's valid Python then :'(

Reply:
oops, yeah it works , just looks a little wonky will clean that up.